common-custom-user-data-maven-extension/pom.xml

```diff
@@ -110,6 +110,19 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>3.2.4</version>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
```

Comment:
I think this should not replace but attach the shaded artifact as an alternative artifact (cf. https://maven.apache.org/plugins/maven-shade-plugin/examples/attached-artifact.html).

Reply:
Oh, that is neat. That way we can benefit from both version. 👍

Reply:
What is the motivation for publishing two variants?
How will this look in practice for our users and our documentation?

Reply:
If we don't add it as an alternative artifact, Maven will use the regular POM with the shaded artifact and download the fat jar and all regular transitive dependencies. In the documentation we could link directly to the fat jar for the aforementioned use cases.